geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java

```diff
@@ -433,18 +457,19 @@ private long iteratorRangeRemove(Iterator<AbstractOrderedSetEntry> scoresIterato
     return entriesRemoved;
   }
 
-  private byte[] zaddIncr(Region<RedisKey, RedisData> region, RedisKey key,
-      List<byte[]> membersToAdd, ZAddOptions options) {
+  private byte[] zaddIncr(Region<RedisKey, RedisData> region, RedisKey key, byte[] member,
+      double increment, ZAddOptions options) {
+    if (Double.isNaN(increment)) {
+      throw new NumberFormatException(ERROR_NOT_A_VALID_FLOAT);
+    }
```

Comment:
This check should be unnecessary, as the ZAddExecutor already checks for NaN increments (assuming the `Coder.bytesToDouble()` method keeps its NaN check that throws an Exception).